FinnUI/Sources/Fullscreen/NotificationCenter/Views/NotificationCenterView.swift

```diff
@@ -0,0 +1,305 @@
+            return nil
+        }
+        
+        let headerView = tableView.dequeue(NotificationCenterHeaderView.self)
+        headerView.delegate = self
+        headerView.configure(with: headerViewModel, inSection: section)
+        return headerView
+    }
+    
+    public func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        guard dataSource?.notificationCenterView(self, segment: selectedSegment, overflowInSection: section) == true else {
+            return nil
+        }
+        
+        let title = dataSource?.notificationCenterView(self, segment: selectedSegment, titleForFooterInSection: section)
+        let footerView = tableView.dequeue(NotificationCenterFooterView.self)
+        footerView.delegate = self
+        footerView.configure(with: title, inSection: section)
+        return footerView
+    }
+}
+
+// MARK: - NotificationCenterHeaderViewDelegate
+extension NotificationCenterView: NotificationCenterHeaderViewDelegate {
+    func notificationCenterHeaderView(_ view: NotificationCenterHeaderView, didSelectSavedSearchButtonInSection section: Int) {
+        delegate?.notificationCenterView(self, segment: selectedSegment, didSelectSavedSearchButtonIn: section)
+    }
+}
+
+// MARK: - NotificationCenterFooterViewDelegate
+extension NotificationCenterView: NotificationCenterFooterViewDelegate {
+    func notificationCenterSectionFooterView(_ footerView: NotificationCenterFooterView, didSelectButtonIn section: Int) {
+        delegate?.notificationCenterView(self, segment: selectedSegment, didSelectFooterButtonInSection: section)
+    }
+}
+
+// MARK: - Private Methods
+private extension NotificationCenterView {
+    func setup() {
+        addSubview(segmentedControl)
+        addSubview(separatorLine)
+        
+        NSLayoutConstraint.activate([
+            segmentedControl.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .spacingM),
+            segmentedControl.topAnchor.constraint(equalTo: topAnchor, constant: .spacingM),
+            segmentedControl.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.spacingM),
+            
+            separatorLine.leadingAnchor.constraint(equalTo: leadingAnchor),
+            separatorLine.topAnchor.constraint(equalTo: segmentedControl.bottomAnchor, constant: .spacingM),
+            separatorLine.trailingAnchor.constraint(equalTo: trailingAnchor),
+            separatorLine.heightAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale)
+        ])
+    }
+    
+    func setupSegmentedControl() {
+        segmentContainers = []
+        
+        guard let dataSource = dataSource else {
+            return
+        }
+        
+        for segment in 0 ..< dataSource.numberOfSegments(in: self) {
+            let title = dataSource.notificationCenterView(self, titleInSegment: segment)
+            let tableView = UITableView.notificationCenterTableView()
```

Comment:
You could perhaps rename this so it is clearer that it creates a new table view, I was a bit unsure first if it was some singleton thing 😛 

`createNotificationCenterTableView()` or something similar would make it more clear.